lib/src/connection.dart

```diff
@@ -72,6 +74,9 @@ class PostgreSQLConnection extends Object
   /// Password for authenticating this connection.
   final String? password;
 
+  /// AuthenticationScheme for authenticating this connection.
+  AuthenticationScheme authenticationScheme = AuthenticationScheme.SCRAM_SHA_256;
```

Comment:
I think there is no reason to make this mutable on the public API, and maybe, there is no need to expose it at all.
- Let's make this private: `AuthenticationScheme _authenticationScheme = AuthenticationScheme.SCRAM_SHA_256;`
- Let's pass this as an additional parameter to `createAuthenticator`, the only caller point will have access to the private field.

Reply:
I think it depends: I don't know if there are actually auth mechanisms, where the user can decide which to use (e.g. md5 vs. plain text (?)). If not, then the `authenticationScheme` variable even doesn't need to be a variable in the `connection` and can be moved to `_PostgreSQLConnectionStateAuthenticating` as the server communicates with the client which auth mechanism to choose.

You're always welcomed to change the code in my PRs, no need to ask if you're sure about something :)